### PR TITLE
Lb issue 14 single release

### LINKED ISSUE #14: Release Maven artifacts and p2 site in the same Maven run
Now that the problem with timestamps in OSGI MANIFEST/feature and no timestamps in POMs might be resolved (https://github.com/eclipse/xtext/issues/2135#issuecomment-1476054632), we could be able to deploy to Sonatype and publish the p2 site in the same Maven run: the JARS on Sonatype and the p2 site would be the same. @szarnekow what I said in https://github.com/eclipse/xtext/issues/2135#issuecomment-1474183272 doesn't seem to be valid anymore ;)

I don't know if I can work on this right now, because it requires a few adjustments:

- we have a lot of projects, and only a few of them are to be published as Maven artifacts. Currently, this is handled by two separate Maven aggregators. If we want to implement this task, there should be a single Maven aggregator. Most POMs should be modified so that Maven deployment is skipped; to keep things clean, I'd propose the introduction of a sub-parent POM (`xtext-no-maven-deploy-parent`?) where the deployment is skipped. This should be inherited from most projects, that is, the ones not to be deployed on Sonatype.
- the composite update site should be created during the Maven build

The second point is not strictly essential: it's just to make sure that, if the p2 publishing fails, the Maven artifacts are not deployed. This avoids situations where Maven artifacts are published but the p2 publishing has failed. Of course, this can be resolved by having a dedicated backup job only for p2 publishing.

### PR DESCRIPTION
**Pull request by LorenzoBettini**
Closes #14

### CODE REVIEW

File: org.eclipse.xtext.full.releng/pom.xml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		<module>../org.eclipse.xtext.common.types</module>
+		<module>../org.eclipse.xtext.ecore</module>
+		<module>../org.eclipse.xtext.xbase</module>
+		<module>../org.eclipse.xtext.xbase.ide</module>
+		<module>../org.eclipse.xtext.xbase.testing</module>
+		<module>../org.eclipse.xtext.builder.standalone</module>
+		<module>../org.eclipse.xtext.generator</module>
+		<module>../org.eclipse.xtext.java</module>
+		<!-- This is required by the xtext-maven-plugin IT -->
+		<module>../org.eclipse.xtext.purexbase</module>
```
Comment (cdietrich) [inline org.eclipse.xtext.full.releng/pom.xml:57]: we also publish this to maven central meanwhile

File: pom.xml [hunk 2]
```diff
@@ -330,6 +346,9 @@
 			</properties>
 			<build>
 				<plugins>
+					<!--
+					For the moment we do without
```
Comment (cdietrich) [inline pom.xml:350]: TODO change the urls once we move this to eclipse org

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -39,14 +39,14 @@ jobs:
...
 
       - name: Build and test (headless with xvfb in Linux)
-        run: xvfb-run mvn clean verify -PuseJenkinsSnapshots -PbuildP2Repository
+        run: xvfb-run mvn clean verify -PuseJenkinsSnapshots
         if: matrix.os == 'ubuntu-latest'
-        working-directory: org.eclipse.xtext.p2.releng
+        working-directory: org.eclipse.xtext.full.releng
 
       - name: Build and test (in other OSes)
         run: mvn clean verify -PuseJenkinsSnapshots
```
Comment (cdietrich) [inline .github/workflows/maven.yml:47]: is the useJenkinsSnapshots profile still used?

File: jenkins/Jenkinsfile-full-deploy-nightly [hunk 4]
```diff
@@ -0,0 +1,43 @@
+pipeline {
+  agent any
+
+  environment {
+    DOWNLOAD_AREA = '/home/data/httpd/download.eclipse.org/modeling/tmf/xtext/monorepo'
+    REPOSITORY_PATH="${DOWNLOAD_AREA}/updates/nightly"
+  }
+
+  tools {
+    maven "apache-maven-3.8.6"
```
Comment (cdietrich) [inline jenkins/Jenkinsfile-full-deploy-nightly:10]: is the .6 intentional?

File: org.eclipse.xtext.maven.plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+To run the integration tests, the `target/.m2test` folder must first be populated with Maven dependencies and with the IT-SNAPSHOT jar of this Maven plugin.
```
Comment (cdietrich) [inline org.eclipse.xtext.maven.plugin/README.md:1]: very nice readme

File: org.eclipse.xtext.full.releng/pom.xml (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.full.releng/pom.xml:57]: Yes, we also publish that.
With the monorepo I think it's not needed anymore to publish that, but we can leave it like that for the moment.
Or did you mean anything else?

File: pom.xml (hunk 2 above)
Comment (LorenzoBettini) [reply pom.xml:350]: You mean all references to the current monorepo?
I left other TODOs concerning that.

File: .github/workflows/maven.yml (hunk 3 above)
Comment (LorenzoBettini) [reply .github/workflows/maven.yml:47]: It's still used for this:

```
${JENKINS_URL}/job/lsp4j/lastStableBuild/artifact/build/maven-repository
```

File: jenkins/Jenkinsfile-full-deploy-nightly (hunk 4 above)
Comment (LorenzoBettini) [reply jenkins/Jenkinsfile-full-deploy-nightly:10]: Well, I copied that part from the current Jenkinsfiles, e.g., https://github.com/eclipse/xtext-core/blob/master/Jenkinsfile#L32

should I change that with a more recent version?

File: org.eclipse.xtext.full.releng/pom.xml (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.full.releng/pom.xml:57]: am not sure about that. but i assume it does not harm to keep publishing it

File: pom.xml (hunk 2 above)
Comment (cdietrich) [reply pom.xml:350]: yes the new location will be `/xtext` as we had discussed before. but therefore sebastian first needs to do the archive step.

File: .github/workflows/maven.yml (hunk 3 above)
Comment (cdietrich) [reply .github/workflows/maven.yml:47]: have to think about that

File: jenkins/Jenkinsfile-full-deploy-nightly (hunk 4 above)
Comment (cdietrich) [reply jenkins/Jenkinsfile-full-deploy-nightly:10]: keep it for now. we can update later

File: org.eclipse.xtext.full.releng/pom.xml (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.full.releng/pom.xml:57]: Let's say that if someone wants to experiment with the xtext-maven-plugin with an existing example Xbase-DSL they can find purexbase on Central :)

**Comment by LorenzoBettini** [comment]
@cdietrich so I'm going to merge

**Review by cdietrich — APPROVED** [review]

**Pull request merged**